doc/feature/restAPI.md

```diff
@@ -0,0 +1,71 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+* _GET /registry_ returns the URIs for all registered objects, together with the
+  supported REST operation (GET or PUT).
+* Each object can be queried for its schema using a GET operation on the URI of
+  that object, suffixed by /schema.
+
+## Examples
+
+* Get a list of all registered objects:
+  * Request: ```GET http://hostname:port/registry```
+  * Response:
+```
+  {
+    "namespace1/foo": ["GET", "PUT"],
+    "namespace2/bar": ["PUT"]
+  }
+```
+* Get JSON schema for a specific object:
+  * Request: ```GET http://hostname:port/lexis/render/lookout/schema```
+  * Response:
+```
+    {
+        "title": "LookOut",
+        "name": "LookOut",
+        "type": "object",
+        "properties": {
+            "matrix": {
+                "type": "array",
+                "items": {
+                    "type": "number"
+                },
+                "minItems": 16,
+                "maxItems": 16,
+                "default": [ 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 ]
+            }
+        },
+        "required": ["matrix"],
+        "additionalProperties": false
+    }
+
+```
```

Comment:
If the schema is really irrelevant, then I would remove it completely from the example because it confuses people that tends to pay attention to details like me.